selectors/views/custom_status.ts

```diff
@@ -3,34 +3,37 @@
 import {getCurrentUser, getUser} from 'mattermost-redux/selectors/entities/users';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 
-import {getMyPreferences} from 'mattermost-redux/selectors/entities/preferences';
-import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
+import {get} from 'mattermost-redux/selectors/entities/preferences';
 import {Preferences} from 'mattermost-redux/constants';
 
+import {createSelector} from 'reselect';
+import {memoizeResult} from 'mattermost-redux/utils/helpers';
+
 import {GlobalState} from 'types/store';
 
-export function getCustomStatus(state: GlobalState, userID?: string) {
-    const user = userID ? getUser(state, userID) : getCurrentUser(state);
-    const userProps = user.props || {};
-    return userProps.customStatus ? JSON.parse(userProps.customStatus) : {};
+export function makeGetCustomStatus() {
+    return memoizeResult((state: GlobalState, userID?: string) => {
+        const user = userID ? getUser(state, userID) : getCurrentUser(state);
+        const userProps = user.props || {};
+        return userProps.customStatus && JSON.parse(userProps.customStatus);
+    });
 }
 
-export function getRecentCustomStatuses(state: GlobalState) {
-    const preferences = getMyPreferences(state);
-    const key = getPreferenceKey(Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_RECENT_CUSTOM_STATUSES);
-    return preferences[key] && preferences[key].value ? JSON.parse(`${preferences[key].value}`) : [];
-}
+export const getRecentCustomStatuses = createSelector(
+    (state: GlobalState) => get(state, Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_RECENT_CUSTOM_STATUSES),
+    (value) => {
+        return value ? JSON.parse(value) : [];
+    },
+);
 
 export function isCustomStatusEnabled(state: GlobalState) {
     const config = getConfig(state);
     return config && config.EnableCustomUserStatuses === 'true';
 }
 
 function showCustomStatusPulsatingDotAndPostHeader(state: GlobalState) {
-    const preferences = getMyPreferences(state);
-    const key = getPreferenceKey(Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_CUSTOM_STATUS_TUTORIAL_STATE);
-    const hasViewedCustomStatusModal = preferences[key] && preferences[key].value === Preferences.CUSTOM_STATUS_MODAL_VIEWED;
-    return !hasViewedCustomStatusModal;
+    const hasViewedCustomStatusModal = get(state, Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_CUSTOM_STATUS_TUTORIAL_STATE);
+    return hasViewedCustomStatusModal !== Preferences.CUSTOM_STATUS_MODAL_VIEWED;
```

Comment:
This looks really suspicious. maybe we need to tweak some variable names here.

Reply:
This was suggested in the PR comments.

Reply:
What does the `hasViewedCustomStatusModal` variable name mean according to you and how is it used here?

Reply:
I have renamed this variable to customStatusTutorialState